File: epitweetr/run_auto.r
# Define functions
main_loop <- function(folder_dir, hadoop_dir, proxy_dir) {
    Sys.setenv(HTTPS_Proxy = proxy_dir)
    Sys.setenv(HADOOP_HOME = hadoop_dir)
    epitweetr_app(folder_dir)
}

detect_loop <-function(folder_dir, hadoop_dir, proxy_dir) {
    Sys.setenv(HTTPS_Proxy = proxy_dir)
    Sys.setenv(HADOOP_HOME = hadoop_dir)
    epitweetr::detect_loop(folder_dir)
}

search_loop <-function(folder_dir, hadoop_dir, proxy_dir) {
    Sys.setenv(HTTPS_Proxy = proxy_dir)
    Sys.setenv(HADOOP_HOME = hadoop_dir)
    epitweetr::search_loop(folder_dir)
}

detect_loop <-function(folder_dir, hadoop_dir, proxy_dir) {
    Sys.setenv(HTTPS_Proxy = proxy_dir)
    Sys.setenv(HADOOP_HOME = hadoop_dir)
    epitweetr::detect_loop(folder_dir)
}

run_epit <- function(folder_dir, hadoop_dir, proxy_dir) {
    #path = os.system(dir)
    library(epitweetr)
    print("it works")
    main_loop(folder_dir, hadoop_dir, proxy_dir)
    print("detect loop starts")
    detect_loop(folder_dir, hadoop_dir, proxy_dir)
    print("search loop starts")
    search_loop(folder_dir, hadoop_dir, proxy_dir)
    print("fs loop start")
    fs_loop(folder_dir, hadoop_dir, proxy_dir)
}

#set paths
epit_folder_dir <- "path"
set_proxy <- "path"
set_hadoop <- "path"

# Run epitweetr
library(epitweetr)
library(callr)
#library(subprocess) this package is not available anymore

#run_epit(folder_dir = epit_folder_dir, hadoop_dir = set_hadoop, proxy_dir = set_proxy)

#callr::r_bg(function(){
#    main_loop(folder_dir = epit_folder_dir, hadoop_dir = set_hadoop, proxy_dir = set_proxy);
#    detect_loop(folder_dir = epit_folder_dir, hadoop_dir = set_hadoop, proxy_dir = set_proxy);
#    search_loop(folder_dir = epit_folder_dir, hadoop_dir = set_hadoop, proxy_dir = set_proxy)
#    })

system(main_loop(folder_dir = epit_folder_dir, hadoop_dir = set_hadoop, proxy_dir = set_proxy), intern = FALSE, ignore.stdout = FALSE, ignore.stderr = FALSE, wait = FALSE)
system(detect_loop(folder_dir = epit_folder_dir, hadoop_dir = set_hadoop, proxy_dir = set_proxy), intern = FALSE, ignore.stdout = FALSE, ignore.stderr = FALSE, wait = FALSE)
system("search_loop(folder_dir = epit_folder_dir, hadoop_dir = set_hadoop, proxy_dir = set_proxy)", intern = FALSE, ignore.stdout = FALSE, ignore.stderr = FALSE, wait = FALSE)

File: epitweetr/run_auto.py
# defined four R sciprts which have a function of executing each loop 

import subprocess
import os
from subprocess import Popen
from dotenv import load_dotenv

load_dotenv()
path = Path(os.getenv("DATA_PATH"))

file = ['main_loop', 'detect_loop', 'search_loop', 'fs_loop']
#subprocess.run()
#shell=True: string- open another terminal, takes more time / shell=False: list - keep one terminal 
for a in file: 
    print(a)
    subprocess.Popen(f'Rscript {path}/run_epit/%s.R' % a, shell=True)
    #process = subprocess.Popen('Rscript my_path/run_epit/%s.R' % a, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE) #stdout: save output into the variable(process)/ stderr: save error into process
    #rc = process.wait()
    #out, err = process.communicate()
    #print(f'output is: {out}')
    #print(f'error is: {err}')
